src/microk8scluster.py

```diff
@@ -220,6 +226,19 @@ def _containerd_env(self, event):
             env.write(configured)
             subprocess.check_call(['systemctl', 'restart', 'snap.microk8s.daemon-containerd.service'])
 
+    def _refresh_channel(self, _):
+        channel = self.model.config['channel']
+        if channel == 'auto':
+            return
+        infostr = subprocess.check_output('snap info microk8s'.split())
+        info = yaml.load(infostr)
```

Comment:
I'd use `yaml.safe_load` here, simply as best practice, although it's certainly had to imagine `snap info` trying to execute such an attack!

Reply:
Done.